sycl/test/basic_tests/parallel_for_disable_range_roundup.cpp

```diff
@@ -0,0 +1,53 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -DSYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING %s -o %t.out
+// RUN: env SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE=1 %RUN_ON_HOST %t.out
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+
+range<1> Range1 = {0};
+
+void check(const char *msg, size_t v, size_t ref) {
+  std::cout << msg << v << std::endl;
+  assert(v == ref);
+}
+
+int try_rounding_off(size_t size) {
+  range<1> Size{size};
+  int Counter = 0;
+  {
+    buffer<range<1>, 1> BufRange(&Range1, 1);
+    buffer<int, 1> BufCounter(&Counter, 1);
+    queue myQueue;
+
+    std::cout
+        << "Run parallel_for with SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING"
+        << std::endl;
+    myQueue.submit([&](handler &cgh) {
+      auto AccRange = BufRange.get_access<access::mode::read_write>(cgh);
+      auto AccCounter = BufCounter.get_access<access::mode::atomic>(cgh);
+      cgh.parallel_for<class PF_init_item1>(Size, [=](item<1> ITEM) {
+        AccCounter[0].fetch_add(1);
+        AccRange[0] = ITEM.get_range(0);
+      });
+    });
+    myQueue.wait();
+  }
+  check("Size seen by user = ", Range1.get(0), size);
+  check("Counter = ", Counter, size);
+  return 0;
+}
+
+int main() {
+  int x;
+
+  x = 10;
+  try_rounding_off(x);
+
+  return 0;
+}
+
+// CHECK:  Run parallel_for with SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING
+// CHECK-NOT: parallel_for range adjusted from 10 to 32
```

Comment:
```suggestion
// CHECK-NOT: parallel_for range adjusted from 10
```
Suggest truncating the line to make the check more robust.
Another approach is to use LIT variables.

Reply:
OK